engine/src/test/java/net/jqwik/api/EMailsTests.java

```diff
@@ -219,48 +220,70 @@ void domainsWithOneAndMorePartsAreGenerated(@ForAll("emails") String email){
 	@Group
 	class ShrinkingTests {
 
-		@Property
+		@Property(tries = 20)
+		void defaultShrinking(@ForAll Random random){
+			Arbitrary<String> emails = Arbitraries.emails();
+			assertAllValuesAreShrunkTo("A@a", emails, random);
+		}
+
+		@Property(tries = 20)
 		void domainShrinking(@ForAll Random random){
-			Arbitrary<String> emails = Arbitraries.emails().filter(v -> domainShrinkingFilter(v));
-			assertAllValuesAreShrunkTo("a@a", emails, random);
+			Arbitrary<String> emails = Arbitraries.emails();
+			Falsifier<String> falsifier = falsifyDomain();
+			String value = shrinkToMinimal(emails, random, falsifier);
+			assertThat(value).isEqualTo("A@a");
 		}
 
-		@Property
+		@Property(tries = 20)
 		void domainShrinkingWithTLD(@ForAll Random random){
-			Arbitrary<String> emails = Arbitraries.emails().filter(v -> domainShrinkingWithTLDFilter(v));
-			assertAllValuesAreShrunkTo("a@a.aa", emails, random);
+			Arbitrary<String> emails = Arbitraries.emails();
+			Falsifier<String> falsifier = falsifyDomainWithTLD();
+			String value = shrinkToMinimal(emails, random, falsifier);
+			assertThat(value).isEqualTo("A@a.aa");
 		}
 
-		@Property
+		@Property(tries = 50)
 		void ipv4Shrinking(@ForAll Random random){
-			Arbitrary<String> emails = Arbitraries.emails().filter(v -> ipv4ShrinkingFilter(v));
-			assertAllValuesAreShrunkTo("a@[0.0.0.0]", emails, random);
+			Arbitrary<String> emails = Arbitraries.emails();
+			Falsifier<String> falsifier = falsifyIPv4();
+			String value = shrinkToMinimal(emails, random, falsifier);
+			assertThat(value).isEqualTo("A@[0.0.0.0]");
 		}
 
-		@Property
+		@Property(tries = 20)
 		void ipv6Shrinking(@ForAll Random random){
-			Arbitrary<String> emails = Arbitraries.emails().filter(v -> ipv6ShrinkingFilter(v));
-			assertAllValuesAreShrunkTo("a@[::0:0:0:0:0:0]", emails, random);
+			Arbitrary<String> emails = Arbitraries.emails();
+			Falsifier<String> falsifier = falsifyIPv6();
+			String value = shrinkToMinimal(emails, random, falsifier);
+			assertThat(value).isEqualTo("A@[::0:0:0:0:0:0]");
 		}
```

Comment:
How can I fix this problem?
`Expecting:
     <"A@[0:0:0:0:0::0:0]">
    to be equal to:
     <"A@[::0:0:0:0:0:0]">
    but was not.`

Reply:
If "[::0:0:0:0:0:0]" is a valid ipv6 address then the shrunk result is correct. So is it?

Reply:
Yes, it is valid

Reply:
Then check that all present parts are "0" and that there are 6 of them. Would that work?

Reply:
My version now consists in checking that each part is '0' or blank and that it is a valid IP address.